### ISSUE TITLE
saffire <command> could scan for commands

### ISSUE THREAD

**Issue by Addvilz** [saffire/saffire#264]
One little thing I noticed in PR #263 is that command path is taken from file. Maybe it would be nice to scan $HOME/.saffire/commands or similar, as well as default command path always, and leave the path in config empty, that can be optionally defined if user so chooses? 

This could give a single extension point for developers to publish commands via other means, for example, .DEB and would give a kind of auto discovery feature for commands.

Thoughts?

**Comment by jaytaph** [comment]
I think somewhere a command path should be hardcoded, just to make sure that whenever no configuration was found (anywhere), at least there is a path configured during configure-time (this could be for instance different based on linux-distribution for instance). So a debian system would always have a dedicated place where these commands reside.

If a user would want to override this, it might create a local configuration (./saffire.ini or ~/saffire.ini maybe) and add the given path there. Or maybe we could introduce a `saffire.extra.command.path` or something, but I'm afraid that if you would get all kind of scattered commands throughout the system if lots of people use their own directories this way for distributing commands?

**Comment by Addvilz** [comment]
Well what I meant is that, we could have 2 hardcoded paths - one in $HOME and one system global (so users can have commands even when not privileged to access global commands), and an optional path configurable in config (perhaps). 

Sorry, for some reason I thought "config.c" was a config file already...

**Comment by jaytaph** [comment]
ah ok..   yeah, maybe that might be possible. I'm not sure what a good strategy is here through. For instance, there are some hardcoded paths (for now) in when looking for saffire modules (for instance, `.`, `./saffire_modules` etc..  something similar could also be done for commands for your situation. Might indeed make sense

**Comment by Addvilz** [comment]
Yep, that was the idea. This would empower both admins and unprivileged users to have a kind of a plug-and-play thingy. 

However, there is a catch - what about versions and bc breaks for commands/modules?
